Zend/zend_language_scanner.l

```diff
@@ -631,6 +634,27 @@ zend_op_array *compile_filename(int type, zval *filename)
 		convert_to_string(&tmp);
 		filename = &tmp;
 	}
+	if (!CG(script_extensions)[0]) {
+		illegal_script_extension = 0;
+	} else {
+		for(i = 0; i < ZEND_MAX_SCRIPT_EXTENSIONS; i++) {
+			if (!CG(script_extensions)[i]) {
+				break;
+			}
+			len = strlen(CG(script_extensions)[i]);
+			if (Z_STRLEN_P(filename) < len) {
+				break;
+			}
+			p = Z_STRVAL_P(filename) + Z_STRLEN_P(filename) - len;
+			if (!memcmp(p, CG(script_extensions)[i], len)) {
```

Comment:
you're using case sensitive match here. Are you sure it won't break on case-insensitive filesystems?